fe/fe-core/src/main/java/org/apache/doris/qe/PointQueryExecutor.java

```diff
@@ -122,16 +148,22 @@ void setScanRangeLocations() throws Exception {
         }
 
         candidateBackends = new ArrayList<>();
+        this.backendId2TabletIds = scanNode.getBackendId2TabletIds();
         for (Long backendID : scanNode.getScanBackendIds()) {
             Backend backend = Env.getCurrentSystemInfo().getBackend(backendID);
             if (SimpleScheduler.isAvailable(backend)) {
                 candidateBackends.add(backend);
+                if (backendId2TabletIds.containsKey(backendID)) {
```

Comment:
why need to retainAll(tabletIDs), tabletIDs must contain all backendId2TabletIds.get(backendID), and the logic is redundant?